mem0-ts/src/oss/src/vector_stores/upstash.ts

```diff
@@ -0,0 +1,284 @@
+import { Index } from "@upstash/vector";
+import { VectorStore } from "./base";
+import { SearchFilters, VectorStoreConfig, VectorStoreResult } from "../types";
+
+interface UpstashConfig extends VectorStoreConfig {
+  restUrl?: string;
+  restToken: string;
+  apiKey?: string;
+  email?: string;
+  indexName: string;
+  embeddingModel?: // these are the models supported by Upstash as of now
+  | "BGE_SMALL_EN_V1_5"
+    | "BGE_BASE_EN_V1_5"
+    | "BGE_LARGE_EN_V1_5"
+    | "BGE_M3"
+    | "BERT_BASE_UNCASED"
+    | "UAE_LARGE_V1"
+    | "ALL_MINILM_L6_V2"
+    | "MXBAI_EMBED_LARGE_V1"
+    | "BM25";
+}
+
+export class UpstashVectorStore implements VectorStore {
+  private client: Index | null = null;
+  private indexName: string;
+  private dimensions: number;
+  private apiKey?: string;
+  private embeddingModel?: string;
+  private email?: string;
+
+  constructor(config: UpstashConfig) {
+    if (config.restUrl && config.restToken) {
```

Comment:
After creating vdb only we get restUrl & restToken to connect, if passed then initialize client or else `this.initialize()` will run using `apiKey` and `email`.

<img width="1247" height="284" alt="image" src="https://github.com/user-attachments/assets/cd0c6578-94ae-4ee8-ae24-358066cb06ac" />